airmozilla/popcorn/renderer.py

```diff
@@ -0,0 +1,92 @@
+import tempfile
+import os
+import timezone
+
+from airmozilla.base.utils import build_absolute_url, prepare_vidly_video_url
+from airmozilla.popcorn.models import PopcornEdit
+from airmozilla.main.models import Event, VidlySubmission
+from airmozilla.manage import vidly
+from airmozilla.uploads.models import Upload
+
+from django.db import transaction
+from django.conf import settings
+
+import boto
+from boto.s3.key import Key
+from funfactory.urlresolvers import reverse
+from popcoder.popcoder import process_json
+
+
+@transaction.atomic
+def render_edit(id):
+    edit = PopcornEdit.objects.get(id=id)
+    event = edit.event
+
+    filename = '%s.webm' % edit.id
+    filepath = os.path.join(tempfile.gettempdir(), filename)
+    filesize = os.stat(filepath)[os.stat.st_size]
+    if not (os.path.isfile(filepath) and filesize > 0):
+        edit.status = PopcornEdit.STATUS_PROCESSING
+        edit.save()
+        process_json(edit.data, out=filepath)
+
+    # Uploads file to s3
+    connection = boto.connect_s3(
+        settings.AWS_ACCESS_KEY_ID,
+        settings.AWES_SECRET_ACCESS_KEY,
+    )
+    bucket = connection.get_bucket('vidly_uploads')
+    video_key = Key(bucket)
+
+    video_key.key = filename
+    video_key.set_contents_from_filename(filepath)
+    video_url = video_key.generate_url(expires_in=0, query_auth=False)
+
+    Upload.objects.create(
+        user=edit.user,
+        url=video_url,
+        file_name=filename,
+        mime_type='video/webm',
+        size=filesize,
+        upload_time=timezone.now()
+    )
+
+    webhook_url = build_absolute_url(reverse('popcorn:vidly_webhook'))
+    token_protection = event.privacy != Event.PRIVACY_PUBLIC
+
+    tag, error = vidly.add_media(
+        prepare_vidly_video_url(video_url),
```

Comment:
The URL we pass to `vidly.add_media` need to match what we pass to `VidlySubmission.objects.create(url=video_url, ...`
So, change it to this:
```python
video_url = prepare_vidly_video_url(video_url)
tag, error = vidly.add_media(
    video_url,
    ...
)
VidlySubmission.objects.create(
    event=event,
    url=video_url,
    ...
)
```